src/components/user/PersonalProfileEditForm.js

```diff
@@ -44,6 +55,23 @@ export default function PersonalProfileEditForm() {
       });
     } else {
       try {
+        // If there's a profile picture, upload it first
+        if (formData.profilePicture) {
+          const formDataImage = new FormData();
+          formDataImage.append('image', formData.profilePicture);
+
+          // Make the API call to upload the image
+          const imageResponse = await axiosInstance.post(
+            '/:userId/uploadProfile',
+            formDataImage
+          );
+
+          // Assuming the backend returns the image URL, update the formData
+          setFormData((prevData) => ({
+            ...prevData,
+            profilePicture: imageResponse.data.imageUrl,
+          }));
```

Comment:
I think we can remove this code. We are already displaying the image when the user selects it from the file; so there's nothing to change here.

In addition, this will currently break things:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/31baf47d-809f-42f0-903e-61601a97afb2)

This error shows up after I successfully upload a profile image. This happens because `imageResponse.data.imageUrl` is a String; but down on line 157, we are calling `URL.createObjectUrl(formData.profilePicture)`. `createObjectURL` is a function that expects to receive a **file** (in javascript called a Blob) but now we're gonna be passing it a String. 

Let's avoid this by just removing these lines altogether for simplicity.

Reply:
I have removed this function , commit ba8b8cdb428068919638872c1554313871677e58
Now the variable imageResponse has eslint error